tests/v2/validation/publicapi/clusterroletemplatebindings/crtb_create_steve_test.go

```diff
@@ -0,0 +1,330 @@
+						clusterName = nameStr
+						break
+					}
+				}
+			}
+		}
+	}
+	return clusterName
+}
+
+func validateCrtbCreationBySteveWithAllRequiredInputValues(t *testing.T, client *rancher.Client, remoteClusterID string, roleTemplateName string) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	crtbName := namegen.RandStringLower(8)
+
+	readJson, err := os.ReadFile(crtbJSONFilePath)
+	require.NoError(t, err)
+
+	var inputData map[string]interface{}
+	err = json.Unmarshal(readJson, &inputData)
+	require.NoError(t, err)
+	inputData["metadata"].(map[string]interface{})["namespace"] = remoteClusterID
+	inputData["clusterName"] = remoteClusterID
+	inputData["metadata"].(map[string]interface{})["name"] = crtbName
+	inputData["userName"] = newUser.ID
+	inputData["roleTemplateName"] = roleTemplateName
+
+	resp, err := client.Steve.SteveType("management.cattle.io.clusterroletemplatebinding").Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+}
+
+func validateCrtbCreationBySteveWithServiceAccountAsInputAndItIsIgnored(t *testing.T, client *rancher.Client, remoteClusterID string) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	crtbName := namegen.RandStringLower(8)
+
+	readJson, err := os.ReadFile(crtbJSONFilePath)
+	require.NoError(t, err)
+
+	var inputData map[string]interface{}
+	err = json.Unmarshal(readJson, &inputData)
+	require.NoError(t, err)
+	inputData["metadata"].(map[string]interface{})["namespace"] = remoteClusterID
+	inputData["clusterName"] = remoteClusterID
+	inputData["metadata"].(map[string]interface{})["name"] = crtbName
+	inputData["userName"] = newUser.ID
+	inputData["serviceAccount"] = "testServiceAccount"
+
+	resp, err := client.Steve.SteveType("management.cattle.io.clusterroletemplatebinding").Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+	assert.NotContains(t, (*resp).JSONResp, "serviceAccount")
+}
+
+func validateCrtbCreationBySteveWithUserPrincipalNameAsInputAndItIsUsed(t *testing.T, client *rancher.Client, remoteClusterID string) {
+	newUser, err := users.CreateUserWithRole(client, users.UserConfig())
+	crtbName := namegen.RandStringLower(8)
+
+	readJson, err := os.ReadFile(crtbJSONFilePath)
+	require.NoError(t, err)
+
+	var inputData map[string]interface{}
+	err = json.Unmarshal(readJson, &inputData)
+	require.NoError(t, err)
+	inputData["metadata"].(map[string]interface{})["namespace"] = remoteClusterID
+	inputData["clusterName"] = remoteClusterID
+	inputData["metadata"].(map[string]interface{})["name"] = crtbName
+	inputData["userName"] = newUser.ID
+	inputData["userPrincipalName"] = "local://" + newUser.ID
+
+	resp, err := client.Steve.SteveType("management.cattle.io.clusterroletemplatebinding").Create(inputData)
+	require.NoError(t, err)
+	assert.Contains(t, (*resp).JSONResp["id"], crtbName)
+	fmt.Printf("Variable i is of type: %T\n", (*resp).JSONResp["userPrincipalName"])
```

Comment:
Is using fmt.Print a good practice? Was this added for debugging purpose? Can we remove this?

Reply:
Please check now, its resolved